src/s3-iodav2-v3.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+from datetime import datetime
+import utils.time_utils as time_utils
+import utils.file_utils as file_utils
+import yaml
+import sys
+import boto3
+import botocore
+import subprocess
+import os
+import pathlib
+
+#----- check the python version because this assumes we use ordered dictionaries
+assert sys.version_info >= (3, 6), "Requires python > 3.6"
+
+#----- load and parse yaml file
+#yamlpath = '../test/testinput/s3_source.yaml'
+assert len(sys.argv) == 2, "Config file not specified. Required calling sequence: s3-copy.py <config.yaml>"
+yamlpath = sys.argv[1]
+file_utils.is_valid_readable_file(yamlpath)
+f = open(yamlpath)
+config_dict = yaml.safe_load(f)
+f.close()
+
+# read date range
+dr = time_utils.get_date_range_from_dict(config_dict.get('date_range'))
+# read cycling interval in seconds
+cycling_interval = config_dict.get('cycling_interval')
+# read source info
+sources = config_dict.get('source')
+# read destination info
+destination = config_dict.get('destination')
+#read "move" flag
+#move_flag = config_dict.get('remove source file','false').lower() == 'true'
+move_flag = config_dict.get('remove source file','false')
+PY_CURRENT_DIR = pathlib.Path(__file__).parent.resolve()
+
+#----- create source clients
+print('Opening aws resources')
+destination_bucket = boto3.Session(profile_name=destination.get('credential_profile')).resource('s3').Bucket(destination["bucket"])
+destination_fn_template = destination.get('path')+destination["file_template"]["prefix"]+destination["file_template"]["root"]+destination["file_template"]["suffix"]
+destionation_file_template = destination["file_template"]["prefix"]+destination["file_template"]["root"]+destination["file_template"]["suffix"]
+source_buckets = []
+for source in sources.values():
+  source_buckets.append( boto3.Session(profile_name=source["credential_profile"]).resource('s3').Bucket(source["bucket"]))
+print('...Done')
+
+#----- open log files
+print('Opening loggers')
+dtg = datetime.now().strftime('%Y%m%d%H%M%S')
+logger_present = open(config_dict["logging"]["present"].format(dtg=dtg),'wt')
```

Comment:
maybe this log file is not needed if the comment below is valid?

Reply:
agreed and removed